dev/CommonStyles/APITests/CommonStylesTests.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Common;
+using MUXControlsTestApp.Utilities;
+using System.Linq;
+using System.Threading;
+using Windows.Foundation.Metadata;
+using Windows.System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using MUXControls.TestAppUtils;
+using PlatformConfiguration = Common.PlatformConfiguration;
+using OSVersion = Common.OSVersion;
+using System.Collections.Generic;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+    [TestClass]
+    public class CommonStylesVisualTreeTestSamples
+    {
+        [TestMethod]
+        public void VerifyVisualTreeForAppBarAndAppBarToggleButton()
+        {
+            if (!PlatformConfiguration.IsOsVersion(OSVersion.Redstone5))
+            {
+                return;
+            }
+
+            var xaml = @"<Grid Width='400' Height='400' xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation' xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'> 
+                            <StackPanel>
+                                <AppBarButton Icon='Accept' Label='Accept'/>
+                                <AppBarToggleButton Icon='Dislike' Label='Dislike'/>
+                            </StackPanel>
+                       </Grid>";
+            VisualTreeTestHelper.VerifyVisualTree(xaml: xaml, 
```

Comment:
Do i need to pass the xaml ? can i create the tree, set it up however i want, change properties etc and then compare ? can i pass it a uielement that is already instantiated ?

Reply:
yes. I ever support that you only dump a visualtree for an existing UI. Here are some function signatures.
```
public static void VerifyVisualTree(UIElement root, string masterFilePrefix, Theme theme = Theme.None, IPropertyValueTranslator translator = null, IFilter filter = null, IVisualTreeLogger logger = null)
 public static void VerifyVisualTree(string xaml, string masterFilePrefix, Theme theme = Theme.None, IPropertyValueTranslator translator = null, IFilter filter = null, IVisualTreeLogger logger = null)
 public static string DumpVisualTree(DependencyObject root, IPropertyValueTranslator translator = null, IFilter filter = null, IVisualTreeLogger logger = null)
```